examples/data-objects/vltava/src/index.ts

```diff
@@ -78,9 +78,13 @@ export class VltavaRuntimeFactory extends ContainerRuntimeFactoryWithDefaultData
     protected async containerHasInitialized(runtime: IContainerRuntime) {
         // Load the last edited tracker fluidObject (done by the setup method below). This fluidObject
         // provides container level tracking of last edit and has to be loaded before any other fluidObject.
-        const tracker = await requestFluidObject<IFluidLastEditedTracker>(
-            await runtime.getRootDataStore(ContainerRuntimeFactoryWithDefaultDataStore.defaultDataStoreId),
-            "");
+        const dataStoreRuntime =
+            await runtime.getRootDataStore(ContainerRuntimeFactoryWithDefaultDataStore.defaultDataStoreId);
+        if (dataStoreRuntime === undefined) {
```

Comment:
The code feels a bit too verbose. I wonder if decision to return undefined from getRootDataStore is good one. (though did not finish looking through PR)

Reply:
Not sure I see where you would change behavior based on undefined. I.e. I feel like (with proper translation between exceptions and responses and try/catch blocks I've added) you can just throw correct error if data store is not there. You might want to use combination of response + responseToException() to ensure that same info is preserved, but otherwise I have not seen the case where you would want to differentiate (maybe I missed it)

Reply:
The difference in behavior is in `ContainerRuntime::resolveHandle`. There, if the `getDataStore` returns undefined, we would default to returning a 404 instead of a 500 that we currently return.
I could throw an error in `getDataStore` and convert that to a response in `resolveHandle`. But currently, `exceptionToResponse` only returns status 500. I can change that to support returning any error code.

Reply:
Also, in general, if someone calls `getRootDataStore` with an id for which we don't even have a data store context, is throwing an error the right thing to do? If the data store is not a root or something goes wrong during realizing it, we should definitely throw. But I was debating whether "not even having the data store" falls into the same bucket.

Reply:
For example, someone might want to call this API to check if a data store exists or not. Wouldn't returning undefined be a better option than throwing?

Reply:
exceptionToResponse() will correctly pick up status code if error was generated via responseToException(), so I think returning 404 is not an issue here.
So I believe getRootDataStore could through proper exception if data store does not exist, and throw any other exception if it hits some other error on the path of loading.

As for checking for existence, I think more explicit API would be warranted. That said, we need to be very careful to add it, as the data it returns is invalid the moment it returns it. Same is true for higher-level constructs (requests), but it's a bit better not to expose it as unique state.

Reply:
Fixed `exceptionToResponse` not translating status code. Also, made request optional in the createResponse methods since they looked optional. Can you please check if that looks good.

Updated getDataStores to throw a response exception which is then correctly converted to a response (including 404 when data store does not exist).